modules/oadp-api-details.adoc

```diff
@@ -0,0 +1,164 @@
+
+|`Configuration`
+|*`ApplicationConfig`
+|Used to define the data protection application's server configuration.
+|===
+
+.BackupLocation
+[options="header"]
+|===
+|Property|Type|Description
+
+|`name`
+|`metav1. ObjectMeta`
+|Location name.
+
+|`velero`
+|*link:(https://velero.io/docs/v1.9/api-types/backupstoragelocation/[velero.BackupStorageLocationSpec]
+|Location to store volume snapshots. For further details, see link:https://github.com/openshift/oadp-operator/blob/076f1495f6de65b0e96392cb6ef7cff66cf7defb/docs/config/bsl_and_vsl.md[Backup Storage Locations and Volume Snapshot Locations Customization].
+|===
+
+.VolumeSnapshot
+[options="header"]
+|===
+|Property|Type|Description
+
+|`name`
+|`metav1. ObjectMeta`
+|Location name.
+
+|`velero`
+|*link:(https://velero.io/docs/v1.9/api-types/volumesnapshotlocation/[velero.VolumeSnapshotLocationSpec]
+|Location to store volume snapshots. For further details, see link:https://github.com/openshift/oadp-operator/blob/076f1495f6de65b0e96392cb6ef7cff66cf7defb/docs/config/bsl_and_vsl.md[Backup Storage Locations and Volume Snapshot Locations Customization].
+|===
+
+.ApplicationConfig (DataProtectionApplicationSpec.Configuration)
+[options="header"]
+|===
+|Property|Type|Description
+
+|`velero`
+|*VeleroConfig
+|Defines the configuration for the Velero server.
+
+|`restic`
+|*resticConfig
+|Defines the configuration for the Restic server.
+|===
+
+.VeleroConfig
+[options="header"]
+|===
+|Property|Type|Description
+
+|`featureFlags`
+|*[] string
+|Defines the list of features to enable for the Velero instance.
+
+|`defaultPlugins`
+|*[] string
+|Five types of default Velero plug-ins can be installed: `AWS`,`Azure`, `CSI`, `GCP`, and `OpenShift`.
+
+|`customPlugins`
+|map [string]interface{}
+|Used for installation of custom Velero plug-ins.
+
+See link:https://github.com/openshift/oadp-operator/blob/076f1495f6de65b0e96392cb6ef7cff66cf7defb/docs/config/plugins.md[plug-ins customization] for further information on `defaultPlugins` and `customPlugins`.
+
+|`restoreResourcesVersionPriority`
+|string
+|Represents a configmap that is created if defined for use in conjunction with the `EnableAPIGroupVersions` feature flag. Defining this field automatically adds `EnableAPIGroupVersions` to the Velero server feature flag.
+
+|`noDefaultBackupLocation`
+|bool
+|if you need to install Velero without a default backup storage location, the `noDefaultBackupLocation` flag must be set in order confirm installation.
+
+|`podConfig`
+|*`PodConfig`
+|Defies the configuration of the Velero pod.
+|===
+
+.ResticConfig
+[options="header"]
+|===
+|Property|Type|Description
+
+|`enable`
+|*bool
+|If set to `true`, enables backup/restore using Restic. If set to `false`, snapshots are needed.
+
+|`supplementalGroups`
+|[]int64
+|Defines the Linux groups to be applied to the Restic pod.
+
+|`timeout`
+|string
+|Defines the Restic timeout. Default value is 1 hour.
+
+
+|`podConfig`
+|*`podConfig`
```

Comment:
maybe add a link *[PodConfig](https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#PodConfig) or hyperlink to `Table 7. PodConfig`